### ISSUE TITLE
Feature Request: deploy VNET and NSG in landing zones and how to structure prod, dev and DR environments

### ISSUE THREAD

**Issue by sudhir-ops** [Azure/terraform-azurerm-caf-enterprise-scale#550]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure caf enterprise scale only deploys certain resources like VWAN,RSG,policies,management groups etc... But How can we create Vnets, subnets and other resources in different landing zones. Are there any modules as part of TF CAF enterprise scale 

#### Is your feature request related to a problem?
NO

#### Describe the solution you'd like

I'm relatively new for terraform CAF enterprise scale. So kindly help me get the answers
We are creating a SAP migration project by leveraging terraform caf enterprise scale module.
I was able to deploy core landing zone,connectivity,identity and management management groups.
Also deployed VWAN and VWAN hub resources. Now I would like to create Vnet and other resources needed for our project in landing zones and link them to VWAN hubs

I no where found documentation on creating vnets,nsg,VM etc in documentation . I would like to know if there are additional modules created for deploying individual resources like vnets, nsg, vm or should I create my own modules and deploy those in respective landing zones. 

Also I would like to know how to structure the prod, non-prod and DR in landing zones. From your videos I came to understanding that the structure should not be as below. In that case , should how can we structure the prod,dev and DR environments.

![how not to deploy landing zone](https://user-images.githubusercontent.com/55252751/207197270-0a162dfa-d797-440c-84f4-f57833b2d119.png)

In short
1. I would like to know if it is possible to deploy individual resources using caf-enterprise scale terraform like we are creating connectivity resources like VWAN
2. How to structure the prod , dev, DR in landing zone hierarchy

**Comment by sudhir-ops** [comment]
Thank you for the detailed explanation. I understood, we might need to leverage other open source modules or creating my own modules. Also learnt how to deploy prod/dev/test workloads as advised

 I'm planning to create a folder structure like below for future development like below
<img width="209" alt="folder structure tfcaf" src="https://user-images.githubusercontent.com/55252751/207403380-abba8f89-ed12-4e2b-b522-5897801fbdb8.PNG">
 
Please let me know if there are any standard folder structure to be followed as part of CAF enterprise scale for adding in new workload resources. If not , I can follow the existing folder structure as above

**Comment by krowlandson** [comment]
@sudhir-ops we have seen other customers using the above approach, and I am actually working on some guidance for how to use the module in this way.

The only comment I have is to note that as we don't deploy any resources into the identity subscription (the config controls policy configuration only), we generally recommend just covering this off as a part of core. Of course you can use a dedicated workspace for identity if you plan to use your own code to deploy resources into your identity subscription 👍🏻 

Hope this helps?

**Comment by sudhir-ops** [comment]
Thank you for the explanation. I will restructure the folders as per your suggestion. You can close this issue.
